cpp_utils/src/cpp/time/time_utils.cpp

```diff
@@ -36,17 +36,25 @@ namespace utils {
 
 Timestamp now() noexcept
 {
-    return std::chrono::system_clock::now();
+    return Timeclock::now();
 }
 
 Timestamp the_end_of_time() noexcept
 {
-    return std::chrono::time_point<std::chrono::system_clock>::max();
+    #if _EPROSIMA_WINDOWS_PLATFORM // In Windows std::gmtime does not support negative values nor values greater than 2^32
+    return Timeclock::from_time_t(std::numeric_limits<long>::max());
+    #else
+    return Timestamp::max();
+    #endif // if _EPROSIMA_WINDOWS_PLATFORM
 }
 
 Timestamp the_beginning_of_time() noexcept
 {
-    return std::chrono::time_point<std::chrono::system_clock>::min();
+    #if _EPROSIMA_WINDOWS_PLATFORM // In Windows std::gmtime does not support negative values nor values greater than 2^32
+    return Timeclock::from_time_t(0);
```

Comment:
Same as above. Apparently, at least in the Windows machine I have been testing, beginning of time actually is "1969-12-31_12-00-00", but I agree better to cut at 0. Mind blowing...